sdk/src/network-client.ts

```diff
@@ -551,16 +588,60 @@ class AleoNetworkClient {
    * const expectedValue = "0u64";
    * assert.equal(mappingValue, expectedValue);
    */
-  async getProgramMappingValue(programId: string, mappingName: string, key: string): Promise<string> {
+  async getProgramMappingValue(programId: string, mappingName: string, key: string | Plaintext): Promise<string> {
     try {
-      return await this.fetchData<string>("/program/" + programId + "/mapping/" + mappingName + "/" + key)
+      const keyString = key instanceof Plaintext ? key.toString() : key;
+      return await this.fetchData<string>("/program/" + programId + "/mapping/" + mappingName + "/" + keyString)
     } catch (error) {
       throw new Error("Error fetching mapping value - ensure the mapping exists and the key is correct");
     }
   }
 
+
   /**
-   * Returns the latest state/merkle root of the Aleo blockchain
+   * Returns the value of a mapping as a wasm Plaintext object. Returning an
+   * object in this format allows it to be converted to a Js type and for its
+   * internal members to be inspected if it's a struct or array.
+   *
+   * @example
+   * // Get the bond state as an account.
+   * const unbondedState = networkClient.getMappingPlaintext("credits.aleo", "bonded", "aleo1rhgdu77hgyqd3xjj8ucu3jj9r2krwz6mnzyd80gncr5fxcwlh5rsvzp9px");
+   *
+   * // Get the two members of the object individually.
+   * const validator = unbondedState.getMember("validator");
+   * const microcredits = unbondedState.getMember("microcredits");
+   *
+   * // Ensure the expected values are correct.
+   * assert.equal(validator, "aleo1u6940v5m0fzud859xx2c9tj2gjg6m5qrd28n636e6fdd2akvfcgqs34mfd");
+   * assert.equal(microcredits, BigInt("9007199254740991"));
+   *
+   * // Get a JS object representation of the unbonded state.
+   * const unbondedStateObject = unbondedState.toObject();
+   *
+   * const expectedState = {
+   *     validator: "aleo1u6940v5m0fzud859xx2c9tj2gjg6m5qrd28n636e6fdd2akvfcgqs34mfd",
+   *     microcredits: BigInt("9007199254740991")
+   * };
+   * assert.equal(unbondedState, expectedState);
+   *
+   * @param {string} programId - The program ID to get the mapping value of (e.g. "credits.aleo")
+   * @param {string} mappingName - The name of the mapping to get the value of (e.g. "account")
+   * @param {string | Plaintext} key - The key of the mapping to get the value of (e.g. "aleo1rhgdu77hgyqd3xjj8ucu3jj9r2krwz6mnzyd80gncr5fxcwlh5rsvzp9px")
+   *
+   * @return {Promise<string>} String representation of the value of the mapping
+   */
+  async getProgramMappingPlaintext(programId: string, mappingName: string, key: string | Plaintext): Promise<Plaintext> {
+    try {
+      const keyString = key instanceof Plaintext ? key.toString() : key;
+      const value = await this.fetchRaw("/program/" + programId + "/mapping/" + mappingName + "/" + keyString);
+      return Plaintext.fromString(JSON.parse(value));
```

Comment:
Why is this using `JSON.parse` instead of `parseJSON`?

Reply:
Because it doesn't decode into the rust object using parseJSON.